packages/tools/webpack-fluid-loader/src/loader.ts

```diff
@@ -417,17 +417,19 @@ async function attachContainer(
 	shouldUseContainerId: boolean,
 ) {
 	// This is called once loading is complete to replace the url in the address bar with the new `url`.
-	const replaceUrl = (resolvedUrl: IResolvedUrl | undefined) => {
+	const replaceUrl = (attachedContainer: IContainer) => {
+		if (attachedContainer.resolvedUrl === undefined) {
```

Comment:
Suggest lifting this assertion out of the function call so the function can take just an `IResolvedUrl` (more scoped and aligns with function name/intent).

Reply:
that would require two checks. the change let me limit it to a single check, as we call this method from multiple places and the container object can change between.

Reply:
Two checks seems fine to me, but if you're against it then I'd go back to taking the `IResolvedUrl | undefined` and throw on `undefined` - this function doesn't need the container and passing over-scoped objects is a bigger problem IMO.

Or even do a non-null assertion at the callsites with an inline comment explaining that post-attach the resolvedUrl is guaranteed to be defined.